File: Java/SimpleArithmeticApp.java
import java.util.Random;
import java.util.Scanner;
public class SimpleArithmeticApp {
public static void main(String[] args) {

Scanner input = new Scanner(System.in);
Random random = new Random();

int score = 0;
int maxQuestions = 10;

System.out.println("Subtraction problem");
System.out.println(" ");
System.out.println("Note: Yoy have 10 subtraction questions.");
System.out.println("You have only 2 attempts per question.\n");

for (int index = 1; index <= maxQuestions; index++) {

    int num1 = random.nextInt(21);
    int num2 = random.nextInt(21);

    if (num2 > num1) {
        int temp = num1;
        num1 = num2;
        num2 = temp;
    }

          
    int correctAnswer = num1 - num2;

    System.out.println("Question " + index + ": What is " + num1 + " - " + num2 + "?");

    boolean answeredCorrectly = false;

    for (int attempt = 1; attempt <= 2; attempt++) {
        System.out.print("Your answer: ");
        int userAnswer = input.nextInt();

        if (userAnswer == correctAnswer) {
            System.out.println("Correct!\n");
            score++;
            answeredCorrectly = true;
            break;
        } else {
            if (attempt == 1) {
                System.out.println("Wrong. Try again.");
            } else {
                System.out.println("Wrong again. Correct answer = " + correctAnswer + "\n");
            }
        }
}
}
}
}

File: JavaScript/SimpleArithmeticApp.js
const input = require('prompt-sync')();

let score = 0;
const maxQuestions = 10;

console.log("Subtraction problem\n");
console.log("Note: You have 10 subtraction questions.");
console.log("You have only 2 attempts per question.\n");

for (let index = 1; index <= maxQuestions; index++) {

    let num1 = Math.floor(Math.random() * 21);
    let num2 = Math.floor(Math.random() * 21);

    if (num2 > num1) {
        let temp = num1;
        num1 = num2;
        num2 = temp;
    }

    let correctAnswer = num1 - num2;

    console.log(`Question ${index}: What is ${num1} - ${num2}?`);

    let answeredCorrectly = false;

    for (let attempt = 1; attempt <= 2; attempt++) {

        let userAnswer = Number(input("Your answer: "));

        if (userAnswer === correctAnswer) {
            console.log("Correct!\n");
            score++;
            answeredCorrectly = true;
            break;
        } else {
            if (attempt === 1) {
                console.log("Wrong. Try again.");
            } else {
                console.log(`Wrong again. Correct answer = ${correctAnswer}\n`);
            }
        }
    }
}

console.log(`Your final score is: ${score}/${maxQuestions}`);
